safir/src/safir/metrics/_event_manager.py

```diff
@@ -100,6 +103,11 @@ async def publish(self, payload: P) -> EventMetadata:
             accessed.
         """
 
+    @property
+    def published(self) -> PublishedList[P]:
+        """A list of published event payloads with some test helpers."""
+        raise NotImplementedError("Only available on a MockEventPublisher.")
+
```

Comment:
I would just omit this definition from the parent class. That moves the error from runtime to be detectable by mypy. Tests can then use an `isinstance` assertion to check that the publisher is of the type they expected and then freely use `.published`. I think that's better than having a property that can fail at runtime after passing a type check.